app/models/message_model.ml

```diff
@@ -210,13 +224,19 @@ let pp ppf { id; user_id; user_name; user_email; creation_date; content } =
     content
 ;;
 
-let validatation ?(content_field = "message_content") name =
+let created_message { creation_content; _ } = creation_content
+let updated_message { creation_content; _ } = creation_content
+let is_created_preview { is_preview; _ } = is_preview
+let is_updated_preview { is_preview; _ } = is_preview
+
+let validatation ?(content_field = "message_content") name fields =
   let open Lib_form in
   let formlet s =
     let+ content = required s content_field not_blank in
-    { creation_content = content }
+    let is_preview = List.exists (fun (k, _) -> k = "Preview") fields in
```

Comment:
And I suggest to test over `String.(equal (lowercase_ascii @@ trim k) "preview")`

Reply:
I don't really understand that stuff. I've tried

```ocaml
let validatation ?(content_field = "message_content") name =
  let open Lib_form in
  let formlet s =
    let+ content = required s content_field not_blank in
    let+ is_preview = optional s "Preview" not_blank in
    { creation_content = content; is_preview = Option.is_some is_preview }
  in
  run ~name formlet
;;
```
(keeping in mind that `Preview` is submitted as a key, probably because I don't really know how to use forms) but then I get this type error:

```
File "app/models/message_model.ml", line 1:
Error: The implementation app/models/message_model.pp.ml
       does not match the interface app/models/.models.objs/byte/models__Message_model.cmi:
        Values do not match:
          val validate_creation :
            ?content_field:string ->
            (string * string) list ->
            update_form Lib_common.Validate.t Lib_common.Try.t
        is not included in
          val validate_creation :
            ?content_field:string ->
            (string * string) list -> update_form Lib_common.Try.t
        The type
          ?content_field:string ->
          (string * string) list ->
          update_form Lib_common.Validate.t Lib_common.Try.t
        is not compatible with the type
          ?content_field:string ->
          (string * string) list -> update_form Lib_common.Try.t
        Type
          update_form Lib_common.Validate.t Lib_common.Try.t =
            (update_form Lib_common.Validate.t, Lib_common.Error.t) result
        is not compatible with type
          update_form Lib_common.Try.t =
            (update_form, Lib_common.Error.t) result
        Type
          update_form Lib_common.Validate.t =
            (update_form, Lib_common.Error.t Preface_core.Nonempty_list.t)
            Preface_stdlib__Validation.t
        is not compatible with type update_form
        File "app/models/message_model.mli", lines 91-94, characters 0-24:
          Expected declaration
        File "app/models/message_model.ml", line 242, characters 4-21:
          Actual declaration
```

Reply:
If you make a sequence of `let+` and `let+` you will _nest_ your data, so you have to use the monoidal product : `let+ x = ... and+ y = ...`. For this function, I suggest this : 

```ocaml
let validatation ?(content_field = "message_content") ?(preview_field = "Preview") name =
  let open Lib_form in
  let formlet s =
    let+ content = required s content_field not_blank
    and+ preview_value = optional s preview_field is_string in
    { creation_content = content; is_preview = Option.is_some preview_value }
  in
  run ~name formlet
```

(Not tested but it should work). 
Is that more clear for you?

Reply:
Oh waow, thanks.